# Question no :1

 What does empirical rule explain?

Answer.) The empirical rule refers to a statistical rule that mentions that all data or information is covered around three standard deviations of the average in a normal distribution. It states that the first standard deviation covers 68% data, the first and second standard deviation covers 95% data, and all the three deviations of the average in normal distribution cover 99.7% data. The empirical rule is also known as the three-sigma rule, that is 68-95-99.7 rule.

# Question no:2

Give two examples each of Dependent event & Independent Event?

Answer.) Events are dependent if the outcome of one event affects the outcome of another.

For example, if you draw two colored balls from a bag and the first ball is not replaced before you draw the second ball then the outcome of the second draw will be affected by the outcome of the first draw.
Second example is getting into a traffic accident is dependent upon driving or riding in a vehicle.

INDEPENDENT EVENTS 
The outcome of one event does not affect the outcome of the other .
If a and b are independent events then the probability of both occurring is 
p(A and B)=p(A)xp(B)

example

Finding a dollar on the street and buying a lottery ticket; finding a dollar isn’t dictated by buying a lottery ticket, nor does buying the ticket increase your chances of finding a dollar.
 Another example is Growing the perfect tomato and owning a cat

# Question no:3

Q3 A firm is starting a delivery service for a new client between 2 points. Since it is a new client, the firm wants to send more consistent delivery boy to deliver the product. Which delivery boy should they select? (Delivery Boy.xlsx). Solve this problem in Python.??

In [5]:
import pandas as pd
import seaborn as sb
ef=pd.read_csv("Delivery Boy.csv")
ef

deliveryboy1  deliveryboy2
0             12            40
1             13            10
2             17            31
3             21            59
4             24            13
5             24            50
6             26            27
7             27            33
8             27            53
9             30            34
10            32            16
11            35            10
12            37            34
13            38            30
14            41            33
15            43            42
16            44            34
17            46            44
18            53            29
19            60            28

In [6]:
ef.mean()

deliveryboy1    32.5
deliveryboy2    32.5
dtype: float64

In [8]:
ef.var()

deliveryboy1    166.157895
deliveryboy2    181.631579
dtype: float64

  Variance of deliveryboy1 is less it means he is less dispersed and more consistent. So they should select deliveryboy1 for delivery.

# Question no - 4

Load Dataset ‘tips’ from Seaborn Library ?

In [6]:
import pandas as pd
import seaborn as sb
df = sb.load_dataset('tips')
df

total_bill   tip     sex smoker   day    time  size
0         16.99  1.01  Female     No   Sun  Dinner     2
1         10.34  1.66    Male     No   Sun  Dinner     3
2         21.01  3.50    Male     No   Sun  Dinner     3
3         23.68  3.31    Male     No   Sun  Dinner     2
4         24.59  3.61  Female     No   Sun  Dinner     4
..          ...   ...     ...    ...   ...     ...   ...
239       29.03  5.92    Male     No   Sat  Dinner     3
240       27.18  2.00  Female    Yes   Sat  Dinner     2
241       22.67  2.00    Male    Yes   Sat  Dinner     2
242       17.82  1.75    Male     No   Sat  Dinner     2
243       18.78  3.00  Female     No  Thur  Dinner     2

[244 rows x 7 columns]

# Question no - 5

Find the Correlation between ‘Total Bill’ column & ‘Tip’ column ?

In [11]:
import numpy as np 
np.corrcoef(df['total_bill'],df['tip'])

array([[1.        , 0.67573411],
       [0.67573411, 1.        ]])

# Question - 6

Perform Two Sample t test between ‘sex’ column & ‘Tip’ column ?

In [12]:
#T test - 2 Independent Sample

#Dependent Variable --> Tip, Independent Variable --> Sex : Two Tail test Assume significance level / alpha = 5% CI = 95% #Unequal Variance

#H0: Mean of Tip (Y) = Mean of Tip  ie  Sex = No Difference / Impact on Tip
#H1: Mean of Tip (Y) is not equal to Mean of Tip i.e sex has impact on Tip


In [12]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

In [13]:
import scipy.stats as stats
from scipy.stats import ttest_ind #Two Independent Variables t test

In [9]:
from sklearn import preprocessing 
  
# label_encoder object knows how to understand word labels. 
label_encoder = preprocessing.LabelEncoder() 
  
# Encode labels in column 'species'. 
df['sex']= label_encoder.fit_transform(df['sex']) 
  
df['sex'].unique() 

array([0, 1])

In [10]:
print(df['sex'])

0      0
1      1
2      1
3      1
4      0
      ..
239    1
240    0
241    1
242    1
243    0
Name: sex, Length: 244, dtype: int32


In [16]:
ttest,p_value=stats.ttest_ind(df['tip'],df['sex'])

In [17]:
p_value

7.455763290496082e-90

In [18]:
if p_value < 0.05:
    print('Reject Null Hypotesis')
else:
    print('Fail to reject Null Hypothesis')

Reject Null Hypotesis


It means sex has impact on Tip


# Question no - 7


Perform Chi Square Test between ‘sex’ column & ‘smoker’ column ?

In [13]:
#Here H0: Sex AND Smoker ARE INDEPENDENT / UNASSOCIATED
#H1: Sex AND Smoker ARE DEPENDENT /ASSOCIATED

In [19]:
df

total_bill   tip  sex smoker   day    time  size
0         16.99  1.01    0     No   Sun  Dinner     2
1         10.34  1.66    1     No   Sun  Dinner     3
2         21.01  3.50    1     No   Sun  Dinner     3
3         23.68  3.31    1     No   Sun  Dinner     2
4         24.59  3.61    0     No   Sun  Dinner     4
..          ...   ...  ...    ...   ...     ...   ...
239       29.03  5.92    1     No   Sat  Dinner     3
240       27.18  2.00    0    Yes   Sat  Dinner     2
241       22.67  2.00    1    Yes   Sat  Dinner     2
242       17.82  1.75    1     No   Sat  Dinner     2
243       18.78  3.00    0     No  Thur  Dinner     2

[244 rows x 7 columns]

In [21]:
import scipy.stats as stats
import pandas as pd
import numpy as np
import seaborn as sns
from scipy.stats import chi2

In [22]:
df_table=pd.crosstab(df['sex'],df['smoker'])
df_table

smoker  Yes  No
sex            
0        33  54
1        60  97

In [23]:
obs_values=df_table.values
obs_values

array([[33, 54],
       [60, 97]], dtype=int64)

In [24]:
val=stats.chi2_contingency(df_table)
val

(0.008763290531773594,
 0.925417020494423,
 1,
 array([[33.15983607, 53.84016393],
        [59.84016393, 97.15983607]]))

In [25]:
no_of_rows=len(df_table.iloc[0:2,0]) #Row Index, Column Index
no_of_rows

2

In [26]:
Expected_value=val[3]
Expected_value

array([[33.15983607, 53.84016393],
       [59.84016393, 97.15983607]])

In [27]:
no_of_cols=len(df_table.iloc[0,0:2]) #Row Index, Column Index
ddof=(no_of_rows-1)*(no_of_cols-1)
ddof

1

In [28]:
chi_square=sum([(o-e)**2/e for o,e in zip(obs_values,Expected_value)])
chi_square

array([0.00119737, 0.00073745])

In [29]:
chi_square_statistic=chi_square[0]+chi_square[1]
chi_square_statistic

0.001934818536627623

In [30]:

p_value=1-chi2.cdf(x=chi_square_statistic,df=ddof)
p_value

0.964915107315732

In [31]:
if p_value < 0.05:
    print('Reject Null Hypotesis')
else:
    print('Fail to reject Null Hypothesis') # x and y are independent (unassociated)

Fail to reject Null Hypothesis


It means Sex and Smoker are independent

# Question no 8:

Adidas sports recently launched discount schemes at their premium and best performing stores in Delhi, Mumbai and Bangalore. Now, the Analytics team wants to Analyze whether there is an impact of discounts and location on the sales. Do they need to keep different discount structure for different cities?

In [32]:
import numpy as np
import pandas as pd
import statsmodels.api as sm
from statsmodels.formula.api import ols #ordinary least square

In [33]:

bf = pd.read_excel('Adidas.xlsx')

In [34]:
bf

discount       city  sales
0        0.1      Delhi    133
1        0.1      Delhi    107
2        0.1      Delhi    114
3        0.1      Delhi    150
4        0.1      Delhi     85
5        0.2      Delhi    130
6        0.2      Delhi     86
7        0.2      Delhi    130
8        0.2      Delhi    112
9        0.2      Delhi    125
10       0.1     Mumbai     84
11       0.1     Mumbai    114
12       0.1     Mumbai    138
13       0.1     Mumbai    155
14       0.1     Mumbai     78
15       0.2     Mumbai    138
16       0.2     Mumbai    119
17       0.2     Mumbai    146
18       0.2     Mumbai    158
19       0.2     Mumbai    137
20       0.1  Bangalore     94
21       0.1  Bangalore    123
22       0.1  Bangalore    108
23       0.1  Bangalore    148
24       0.1  Bangalore     93
25       0.2  Bangalore    120
26       0.2  Bangalore    127
27       0.2  Bangalore    150
28       0.2  Bangalore    106
29       0.2  Bangalore    127

In [36]:
#Three Hypothesis in two way Anova 
#H0: Population Means of first factors are equal 
#H1: Population Means of second factors are equal 
#H2: There is no interaction between two factors 
#ie two set of factors are Independent

In [35]:
#perform two-way ANOVA
model = ols('sales ~ C(city) + C(discount) + C(discount):C(city)', data=bf).fit()
sm.stats.anova_lm(model, typ=2)

sum_sq    df         F    PR(>F)
C(city)                488.066667   2.0  0.477997  0.625801
C(discount)           1165.633333   1.0  2.283168  0.143840
C(discount):C(city)    911.666667   2.0  0.892857  0.422654
Residual             12252.800000  24.0       NaN       NaN

city: p-value = .625801

discount: p-value = .143840

discouny*city: p-value = .422654

Since the p-values for city(.625801) and discount (..143840) are both greator than .05, this means that both factors do not have a statistically significant effect on sales

#And since the p-value for the interaction effect (.422654) is not less than .05, this tells us that there is no significant interaction effect between city exposure and discount frequency.They are independent.

So they do not need to keep diiferent discounts structures for different cities as it does not have any effect on sales.

# Question no 9

A Business School have launched some mock interviews that claims that it will help students to get more job offers. School's record average of job offers received by students before and after the mock interview for 10 students for 24 weeks. Test the hypothesis that the student has a positive impact of mock interview with 95% confidence level. (Mock Interview.xlsx). Assume alpha=0.05

In [11]:
#H0:Mean of Jobs recieved before = Mean of Jobs recieved after (No Difference) 
#H1:Mean of  Jobs recieved beforeis not equal to Jobs recieved after

In [37]:
import pandas as pd
af = pd.read_excel('Mock Interview.xlsx')
af

id  before  after
0   1       1      3
1   2       0      2
2   3       2      3
3   4       3      4
4   5       2      2
5   6       1      2
6   7       3      4
7   8       2      3
8   9       0      1
9  10       2      3

In [38]:
import numpy as np 
np.mean(af['before'])

1.6

In [39]:
np.mean(af['after'])

2.7

In [40]:
import scipy.stats as stats
from scipy.stats import ttest_rel

In [41]:
ttest,p_value=stats.ttest_rel(af['before'],af['after'])

In [42]:
p_value

0.00017327723687816783

In [43]:
if p_value < 0.05:
    print('Reject Null Hypotesis')
else:
    print('Fail to reject Null Hypothesis')

Reject Null Hypotesis


We have rejeected null hypotesis that means. mean of jobs receieved before and after are not equal.Mean of Job offered is increasing after Mock interview it means , Mock interview has positive impact.

# Question no - 10

Honda Automotive is planning to launch three cars in a new state. The research team conducted a survey in the state to know the buying preferences of the people. They conduct a survey on random sample of people from urban, rural and semi urban areas. Now, Honda Automotive wants to know whether there is a specific preference for cars in different areas? (Honda.xlsx). Assume alpha = 0.05

H0: Car type and Area type are indepenent
H1: Car type and Area type are associated

In [44]:
import scipy.stats as stats
import pandas as pd
import numpy as np
import seaborn as sns
from scipy.stats import chi2

In [45]:
gf=pd.read_excel('Honda.xlsx')

In [46]:
gf

cartype areatype  Unnamed: 2  Unnamed: 3  Unnamed: 4  Unnamed: 5  \
0    honda city    urban         NaN         NaN         NaN         NaN   
1    honda city    urban         NaN         NaN         NaN         NaN   
2    honda city    urban         NaN         NaN         NaN         NaN   
3    honda city    urban         NaN         NaN         NaN         NaN   
4    honda city    urban         NaN         NaN         NaN         NaN   
..          ...      ...         ...         ...         ...         ...   
321   honda crv    rural         NaN         NaN         NaN         NaN   
322   honda crv    rural         NaN         NaN         NaN         NaN   
323   honda crv    rural         NaN         NaN         NaN         NaN   
324   honda crv    rural         NaN         NaN         NaN         NaN   
325   honda crv    rural         NaN         NaN         NaN         NaN   

     Unnamed: 6  Unnamed: 7  Unnamed: 8  Unnamed: 9  Unnamed: 10  
0           NaN         NaN         NaN         NaN          NaN  
1           NaN         NaN         NaN         NaN          NaN  
2           NaN         NaN         NaN         NaN          NaN  
3           NaN         NaN         NaN         NaN          NaN  
4           NaN         NaN         NaN         NaN          NaN  
..          ...         ...         ...         ...          ...  
321         NaN         NaN         NaN         NaN          NaN  
322         NaN         NaN         NaN         NaN          NaN  
323         NaN         NaN         NaN         NaN          NaN  
324         NaN         NaN         NaN         NaN          NaN  
325         NaN         NaN         NaN         NaN          NaN  

[326 rows x 11 columns]

In [47]:
gh_table=pd.crosstab(gf['cartype'],gf['areatype'])

In [48]:
gh_table

areatype      rural  semiurban  urban
cartype                              
honda accord     12         21     29
honda city       48         52     74
honda crv        39         34     17

In [49]:
obs_value = gh_table.values
obs_value

array([[12, 21, 29],
       [48, 52, 74],
       [39, 34, 17]], dtype=int64)

In [50]:
val=stats.chi2_contingency(gh_table)
val

(20.123744875840764,
 0.00047207713790297935,
 4,
 array([[18.82822086, 20.34969325, 22.82208589],
        [52.8404908 , 57.11042945, 64.04907975],
        [27.33128834, 29.5398773 , 33.12883436]]))

In [51]:
Expected_value=val[3]

In [52]:
Expected_value

array([[18.82822086, 20.34969325, 22.82208589],
       [52.8404908 , 57.11042945, 64.04907975],
       [27.33128834, 29.5398773 , 33.12883436]])

In [53]:
alpha=0.05

In [54]:
ddof = (3-1)*(3-1)

In [55]:
chi_square=sum([(o-e)**2/e for  o,e in zip(obs_value,Expected_value)])
chi_square

array([ 7.9015247 ,  1.15149797, 11.0707222 ])

In [56]:
chi_square=sum([(o-e)**2/e for o,e in zip(obs_value,Expected_value)])
chi_square

array([ 7.9015247 ,  1.15149797, 11.0707222 ])

In [57]:
chi_square_statistic = chi_square[0] + chi_square[1] + chi_square[2]

In [58]:
chi_square_statistic

20.123744875840764

In [59]:
p_value=1-chi2.cdf(x=chi_square_statistic,df=ddof)

In [60]:
p_value

0.000472077137902982

In [61]:
if p_value < 0.05 :
    print("Reject Null Hypothesis")
else:
    print("Fail to Reject Null Hypothesis")

Reject Null Hypothesis


It means  Car type and Area type are dependent and associated.So Yes there is specific preference for cars in different area

# Submitted by - Hemant Singh Dangi